File: src/easy/two-sum/two-sum.ts
export function twoSum(nums: number[], target: number): number[] {
    type Num = number;
    type Index = number;
    let seen: Record<Num, Index> = {};

    for (let i = 0; i < nums.length; ++i) {
        let num = nums[i];
        let diff = target - num;

        if (diff in seen) {
            return [seen[diff], i];
        }

        seen[num] = i;
    }

    throw new Error("solution not found");
}

File: src/easy/two-sum/TwoSum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var seen: [Int: Int] = [:]
        for (i, n) in nums.enumerated() {
            if let seenIndex = seen[target - n] {
                return [seenIndex, i]
            }
            seen[n] = i
        }
        fatalError("solution not found")
    }
}
